public/app/core/logs_model.ts

```diff
@@ -326,3 +325,141 @@ export function makeSeriesForLogs(rows: LogRowModel[], intervalMs: number): Time
     };
   });
 }
+
+export function seriesDataToLogsModel(seriesData: SeriesData[], intervalMs: number): LogsModel {
+  const metricSeries: SeriesData[] = [];
+  const logSeries: SeriesData[] = [];
+
+  for (const series of seriesData) {
+    if (series.name !== 'logs') {
+      metricSeries.push(series);
+      continue;
+    }
+
+    if (getFirstTimeField(series) === -1) {
+      continue;
+    }
+
+    if (!hasFieldNamed(series, 'message')) {
+      continue;
+    }
```

Comment:
Do we need to require that the name is 'message' can it just be the first/longest string?  or something like that?

Reply:
Fixed